api/middleWares/zipExtractor.js

```diff
@@ -0,0 +1,22 @@
+import AdmZip from "adm-zip";
+/**
+ * Extracts files from a ZIP buffer.
+ * @param {Buffer} zipBuffer - The ZIP file buffer.
+ */
+
+export const zipExtractor = (zipBuffer) => {
+	try {
+		const zip = new AdmZip(zipBuffer); // Load ZIP from memory
+		const zipEntries = zip.getEntries(); // Get list of files/folders in ZIP
+
+		const extractedFiles = zipEntries.map((entry) => ({
+			name: entry.entryName,
+			isDirectory: entry.isDirectory,
+			content: !entry.isDirectory ? entry.getData().toString() : null,
```

Comment:
```suggestion
			content: entry.isDirectory ? null : entry.getData().toString(),
```

Always avoid negation. Negation is one of the most confusing things that can be added to code. The human brain is very bad at understanding it so it's always, always better to replace

```
if (!thing) doWhenNegative()
else doWhenPositive()
```

with 

```
if (thing) doWhenPositive()
else doWhenNegative()
```